internal/provisioner/cluster.go

```diff
@@ -84,30 +87,49 @@ func CreateCluster(provider, size string, logger log.FieldLogger) error {
 func DeleteCluster(clusterId string, logger log.FieldLogger) error {
 	logger = logger.WithField("cluster", clusterId)
 
-	logger.Info("deleting cluster")
-
 	s3StateStore := "dev.cloud.mattermost.com"
 	dns := fmt.Sprintf("%s-kops.k8s.local", clusterId)
 
-	// Temporarily look for the kops output directory as a local folder named `tmp`. See above.
-	outputDir := "tmp"
+	// Temporarily look for the kops output directory as a local folder named after
+	// the cluster ID. See above.
+	outputDir := fmt.Sprintf("cluster-%s", clusterId)
+
+	// Validate the provided cluster ID before we alter state in any way.
+	logger.Info("verifying cluster ID")
+	_, err := os.Stat(outputDir)
+	if err != nil {
+		return fmt.Errorf("failed to find cluster directory %q: %s", outputDir, err)
+	}
 
 	terraform := terraform.New(outputDir, logger)
 	defer terraform.Close()
-	err := terraform.Init()
+	out, err := terraform.Output("cluster_name")
```

Comment:
Is it true that we'll always be able to `Output` before `Init`? In general, the `Init` is redundant anyway given the local store, but I'm wondering if we should pattern this after the future S3 code anyway, where we'll need to `Init` again after fetching the information locally. Maybe it would break then?

Reply:
Yep! Good point.